manifests/server.pp

```diff
@@ -465,12 +470,16 @@
   $use_legacy_auth_conf      = $::puppet::server_use_legacy_auth_conf,
 ) {
 
+  if $server_facts != $server_foreman_facts {
```

Comment:
Yeah, it would be way better to just remove it entirely. I've done this already in a [separate branch](https://github.com/earsdown/puppet-puppet/tree/remove-server_facts) and am working on tests. Can we skip deprecation and just remove it? Or would you prefer I set it to undef and warn only if set?